File: 975-RangeSumOfBst/975-RangeSumOfBst.py
# Last updated: 8/5/2025, 2:56:26 PM
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: TreeNode, L: int, R: int) -> int:
        path = []

        def rangeSumBSTHelper(root, L, R, path):
            if root is None:
                return
            if root.val >= L and root.val <= R:
                path.append(root.val)

            rangeSumBSTHelper(root.left, L, R, path)
            rangeSumBSTHelper(root.right, L, R, path)

        rangeSumBSTHelper(root, L, R, path)
        print(sum(path))
        return sum(path)

File: 975-RangeSumOfBst/975-RangeSumOfBst.c
// Last updated: 8/5/2025, 2:56:23 PM
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int sum;
int rangeSumBST(struct TreeNode* root, int low, int high){
    sum = 0;
    dfssearch(root, low, high); 
    return sum;
}

void dfssearch(struct TreeNode* root, int low, int high) {
    if (root != NULL) {
        if (root->val >= low && root->val <= high) {
            sum += root->val;
        }
        if (low < root->val) {
            dfssearch(root->left, low, high);
        }
        if (high > root->val) {
            dfssearch(root->right, low, high);
        }
    }
}